lib/pushmi_pullyu/queue.rb

```diff
@@ -0,0 +1,53 @@
+require 'redis'
+
+# Quick and dirty take on sorted sets
+
+# 1) Create a sorted set in Redis (https://redis.io/topics/data-types). Call it preservation_queue
+#
+# 2) In GenericFile add an after_save that:
+#    - determines a monotonically increasing "score". Obvious scores would be either the time in seconds/milliseconds
```

Comment:
It could be pretty arbitrary, as long as it increases (or decreases) over time to give some ordering to the queue.

I'm using epoch as float in this example, which gives us separation between events on the millisecond resolution. We could honestly just use the integer version, which would give us second resolution, with no real difference -- even if multiple events are added on the same second, this scheme still works because it doesn't really matter to us what order things come out of the queue at as long as things that get updated again "later" move to the back of the queue.